taca/utils/bioinfo_tab.py

```diff
@@ -154,22 +169,28 @@ def get_ss_projects(run_dir):
     else:
         csvf=open(FCID_samplesheet_origin, 'rU')
         data=DictReader(csvf)
-    
-
-
 
     for d in data:
         for v in d.values():
-            matches=pattern.search(v)
+            #if project is found
+            matches=proj_pattern.search(v)
             if matches:
-                project_ids.add(matches.group(1))
-    
-    return project_ids
-
-
-
-
-    
-
-    
-
+                projects = matches.group(1)
+                newData = True
+            #if a lane is found
+            matches=lane_pattern.search(v)
+            if matches:
+                lanes = matches.group(1)
+                newData = True
+            #if a sample is found
+            matches=sample_pattern.search(v)
+            if matches:
+                samples = matches.group(1)
+                newData = True
+         
+        #Populates structure and adds FC  to sample status  
+        if newData == True and 'samples' in locals() and 'lanes' in locals():
+            proj_tree[projects][FCID][lanes][samples]
+            proj_tree[projects][FCID][lanes][samples].value = get_status(run_dir)
```

Comment:
why don't you just use `proj_tree[projects][FCID][lanes][samples] = get_status(run_dir)` ?